pymongo/mongo_client.py

```diff
@@ -785,7 +785,11 @@ def __init__(
                 opts = res["options"]
                 fqdn = res["fqdn"]
             else:
-                seeds.update(uri_parser.split_hosts(entity, port))
+                nodes = uri_parser.split_hosts(entity, port)
+                for host in [node[0] for node in nodes]:
+                    if uri_parser._detect_external_db(host):
+                        break
```

Comment:
Can you add a test case for the "break" logic? ie that we only log once even when multiple hosts are given?

Reply:
Good catch, will do!